src/engraving/dom/tie.cpp

```diff
@@ -182,13 +182,19 @@ void TieSegment::editDrag(EditData& ed)
 
 void TieSegment::computeMidThickness(double tieLengthInSp)
 {
-    m_midThickness = style().styleMM(Sid::SlurMidWidth) - style().styleMM(Sid::SlurEndWidth);
-    if (staff()) {
-        m_midThickness *= staff()->staffMag(tie()->tick());
-    }
-    static constexpr double shortTieLimit = 2;
-    if (tieLengthInSp < shortTieLimit) {
-        m_midThickness *= sqrt(tieLengthInSp / shortTieLimit);
+    const double mag = staff() ? staff()->staffMag(tie()->tick()) : 1.0;
+    const double minTieLength = mag * style().styleS(Sid::MinTieLength).val();
+    const double shortTieLimit = mag * 5.0;
+    const double minTieThickness = mag * (0.15 * spatium() - style().styleMM(Sid::SlurEndWidth));
+    const double normalThickness = mag * (style().styleMM(Sid::SlurMidWidth) - style().styleMM(Sid::SlurEndWidth));
+
+    if (tieLengthInSp > shortTieLimit) {
+        m_midThickness = normalThickness;
+    } else {
+        const double A = 1 / (shortTieLimit - minTieLength);
```

Comment:
You should check if shortTieLimit and minTieLength are equal

Reply:
@RomanPudashkin this is ok now, I've put the check a few lines above :)